controller/controller.go

```diff
@@ -16,10 +17,20 @@ type Controller struct {
 	ApiClient contrail.ApiClient
 }
 
-func NewController(ip string, port int) *Controller {
+func NewController(ip string, port int) (*Controller, error) {
 	client := &Controller{}
 	client.ApiClient = contrail.NewClient(ip, port)
-	return client
+
+	// TODO: use environment variables for keystone auth
+	keystone := contrail.NewKeystoneClient("http://10.7.0.54:5000/v2.0", "admin", "admin",
```

Comment:
I think that you should use here not hard-coded values, but variables. Also, add 'JW-66' to TODO comment.